### ISSUE TITLE
[BUG] pdist: every feature beyond the 1024th is silently dropped from the distances

### ISSUE THREAD

**Issue by truong-v** [FlagOpen/FlagGems#5233]
## Summary

The `pdist` wrapper caps the feature tile at `BLOCK_M = min(next_power_of_2(M), 1024)`, and each `_pdist_forward_*_kernel` loads the feature axis exactly once, as a single tile — there is no loop over further tiles. For inputs with more than 1024 features, features `1024 .. M-1` simply never enter the sum: the result is a distance over a truncated feature vector, returned as if it were the full one. No error, no warning. The pattern repeats in all four kernels of the file (`p2`, `p1`, `inf`, `general`); the newer `_pdist_forward.py` in the same directory accumulates over tiles correctly.

Any transformer-scale embedding width (2048, 4096, ...) exceeds the cap. `tests/test_pdist.py` never exceeds 256 features.

The kernels were added in #3427.

## Reproduction

```python
import torch

import flag_gems

torch.manual_seed(0)

# Smallest failing case: 1025 features, one past the tile cap.
x = torch.randn(2, 1025, device="cuda")
ref = torch.pdist(x)
with flag_gems.use_gems():
    res = torch.pdist(x)
trunc = torch.pdist(x[:, :1024].contiguous())
print(f"(2, 1025):  torch.pdist={ref.item():.6f}  flag_gems={res.item():.6f}")
print(f"torch.pdist over only the first 1024 features = {trunc.item():.6f}"
      f"  -> bit-identical to flag_gems: {torch.equal(res, trunc)}")

# Ordinary embedding widths.
for N, M in [(64, 2048), (64, 4096)]:
    x = torch.randn(N, M, device="cuda")
    ref = torch.pdist(x)
    with flag_gems.use_gems():
        res = torch.pdist(x)
    print(f"({N}, {M}): max|res - ref|={(res - ref).abs().max().item():.4f}  "
          f"(ref values ~{ref.mean().item():.1f})")
```

```
(2, 1025):  torch.pdist=45.423508  flag_gems=45.420292
torch.pdist over only the first 1024 features = 45.420292  -> bit-identical to flag_gems: True
(64, 2048): max|res - ref|=21.7262  (ref values ~63.9)
(64, 4096): max|res - ref|=48.2020  (ref values ~90.5)
```

The result at 1025 features is bit-identical to eager `torch.pdist` over only the first 1024 features, so feature 1024 is provably absent from it.

## Environment

- FlagGems master (75cf3ea)
- NVIDIA L40S
- torch 2.6.0+cu124, triton 3.2.0
